### ISSUE TITLE
golang influx client missing

### ISSUE THREAD

**Issue by cblomart** [influxdata/influxdb#11035]
In golang we used the "github.com/influxdata/influxdb/client/v2" package to have a influx client in golang...
It seems that the client is now missing from the repository...

No client, no doc, no info on how to transition...

```
go get github.com/influxdata/influxdb/client/v2
package github.com/influxdata/influxdb/client/v2: cannot find package "github.com/influxdata/influxdb/client/v2" in any of:
/usr/local/go/src/github.com/influxdata/influxdb/client/v2 (from $GOROOT)
/go/src/github.com/influxdata/influxdb/client/v2 (from $GOPATH)
```

**Comment by gihankarunarathne** [comment]
Yes @cblomart. For me, this's kind of migration without thinking about much details.
It may be better to move for a `monorepo`, but should plan better than this.
The docs are still pointing to invalid resources https://docs.influxdata.com/influxdb/v1.7/tools/api_client_libraries/#go.
My docker builds got stuck because of this. Where's the golang client code?

**Comment by winash** [comment]
Same issue with our build as well.

**Comment by timhallinflux** [comment]
You may have missed the announcement that we were switching over Master from InfluxDB v1 to v2.

You can find the golang client libs here: https://github.com/influxdata/influxdb/tree/1.8/client/v2

**Comment by pauldix** [comment]
The 2.0 client will live in its own repository, which should help with some of this. Sorry for the inconvenience everyone, we tried to let people know, but it's hard to get the word out.

**Comment by gihankarunarathne** [comment]
@timhallinflux how do you propose to migrate our code/docker builds to work again with your changes.
Since `RUN go get -u github.com/influxdata/influxdb/client/v2` command inside a `Dockerfile` always look into the default branch on the GitHub as mentioned [here](https://stackoverflow.com/a/42761887/1461060).

I also tried with [gopkg.in](http://labix.org/gopkg.in). Since your packages are inside `/client/v2` folder, that isn't working either.
Even I manage to install `influxdb/client/v2` via some hack, it has internal dependencies which redirect back to `github.com/influxdata/influxdb/client/v2` failed the docker build.
@timhallinflux  Basically, I just want to know how can I install this dependency inside a Docker file?

As @pauldix mentioned, it needs own repo to resolve this. 👍

**Comment by gihankarunarathne** [comment]
@pauldix nope. As you suggested, I added the following line into `Dockerfile`
```
RUN go get -u github.com/influxdata/influxdb/tree/1.7/client/v2
```
Output:
```
Sending build context to Docker daemon  147.5kB
Step 1/9 : FROM golang:1.11.2-alpine
 ---> 57915f96905a
Step 2/9 : WORKDIR /go/src/app
 ---> Using cache
 ---> 02bc78123ae0
Step 3/9 : RUN apk update && apk add git
 ---> Using cache
 ---> b1110053104d
Step 4/9 : RUN go get -u github.com/kataras/iris
 ---> Using cache
 ---> 33a465cd814f
Step 5/9 : RUN go get -u github.com/influxdata/influxdb/tree/1.7/client/v2
 ---> Running in 3ef7686023f8
package github.com/influxdata/influxdb/tree/1.7/client/v2: cannot find package "github.com/influxdata/influxdb/tree/1.7/client/v2" in any of:
	/usr/local/go/src/github.com/influxdata/influxdb/tree/1.7/client/v2 (from $GOROOT)
	/go/src/github.com/influxdata/influxdb/tree/1.7/client/v2 (from $GOPATH)
The command '/bin/sh -c go get -u github.com/influxdata/influxdb/tree/1.7/client/v2' returned a non-zero code: 1
```

**Comment by pauldix** [comment]
Ok, I'm working on pulling client into its own repo and fixing up all the imports in to so that everything is self contained. More very soon...

**Comment by gihankarunarathne** [comment]
> @gihankarunarathne Would pulling it into its own repo fix this for everyone?

@pauldix as far as I tried today, it should work. But you may need to fix other dependencies as well - `https://github.com/influxdata/influxdb/blob/1.7/client/v2/client.go#L20`.

Another suggestion is, add `/client/v2` folder into the `https://github.com/influxdata/influxdb` for the moment (since you're moving towards a `monorepo` ). Longterm solution is going for a different repo.

**Comment by pauldix** [comment]
Ok, I put something here that has fixed up references. Tests pass so I think it should be good. Let me know if this works for you: https://github.com/influxdata/influxdb1-client

**Comment by gihankarunarathne** [comment]
Yes, it worked @pauldix !
```
Sending build context to Docker daemon  147.5kB
Step 1/9 : FROM golang:1.11.2-alpine
 ---> 57915f96905a
Step 2/9 : WORKDIR /go/src/app
 ---> Using cache
 ---> 02bc78123ae0
Step 3/9 : RUN apk update && apk add git
 ---> Using cache
 ---> b1110053104d
Step 4/9 : RUN go get -u github.com/kataras/iris
 ---> Using cache
 ---> 33a465cd814f
Step 5/9 : RUN go get -u github.com/influxdata/influxdb1-client/v2
 ---> Using cache
 ---> 8d5ebb3442b0
Step 6/9 : COPY ./src .
 ---> 220e75b9fafd
```
I'm worndering why did you name the repo as `influxdb1-client`. `influxdb-client` looks fine for me. Just a suggestion!

Anyway thanks a lot for the quick action! 👍

**Comment by satishdotpatel** [comment]
Same issue here, i am using CentOS 7.5

# go get -u github.com/influxdata/influxdb/client/v2
package github.com/influxdata/influxdb/client/v2: cannot find package "github.com/influxdata/influxdb/client/v2" in any of:
	/usr/lib/golang/src/github.com/influxdata/influxdb/client/v2 (from $GOROOT)
	/root/go/src/github.com/influxdata/influxdb/client/v2 (from $GOPATH)

**Comment by gihankarunarathne** [comment]
@satishdotpatel look into https://github.com/influxdata/influxdb/issues/11035#issuecomment-453783909

**Comment by florentinolim** [comment]
Estou com o mesmo erro também, alguém conseguiu resolver?
CentOS Linux release 7.6.1810 (Core)
[root@mq gowork]# go get -u github.com/influxdata/influxdb/client/v2
package github.com/influxdata/influxdb/client/v2: cannot find package "github.com/influxdata/influxdb/client/v2" in any of:
        /usr/lib/golang/src/github.com/influxdata/influxdb/client/v2 (from $GOROOT)
        /opt/gowork/src/github.com/influxdata/influxdb/client/v2 (from $GOPATH)
[root@mq gowork]#

**Comment by gihankarunarathne** [comment]
Refer to https://github.com/influxdata/influxdb/issues/11035#issuecomment-453783909
```
go get -u github.com/influxdata/influxdb1-client/v2
```

**Comment by karalabe** [comment]
@pauldix Errm, you didn't put the latest v2 client into https://github.com/influxdata/influxdb1-client... my december fixup (https://github.com/influxdata/influxdb/pull/10130) wasn't ported over.

**Comment by pauldix** [comment]
Added it in this PR https://github.com/influxdata/influxdb1-client/pull/10. Will have someone review tomorrow and merge.

**Comment by maxp-edcast** [comment]
Where are the docs for github.com/influxdata/influxdb/client/v2 ?

**Comment by timhallinflux** [comment]
@maxp-edcast https://github.com/influxdata/influxdb-client-go
This isn't quite ready for use...

**Comment by mback2k** [comment]
@pauldix It seems like the following does not work inside go projects with go modules enabled, since the repo has no v2 branch or tag:
```
go get github.com/influxdata/influxdb1-client/v2
go: finding github.com/influxdata/influxdb1-client/v2 latest
go: github.com/influxdata/influxdb1-client/v2@v2.0.0-20190402204710-8ff2fc3824fc: missing github.com/influxdata/influxdb1-client/go.mod and .../v2/go.mod at revision 8ff2fc3824fc
go: error loading module requirements
```
Even specifying the master branch does not work:
```
go get github.com/influxdata/influxdb1-client/v2@master
go: finding github.com/influxdata/influxdb1-client/v2 master
go: github.com/influxdata/influxdb1-client/v2@v2.0.0-20190402204710-8ff2fc3824fc: missing github.com/influxdata/influxdb1-client/go.mod and .../v2/go.mod at revision 8ff2fc3824fc
go: error loading module requirements
```
Please take a look at: https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher
